### ISSUE TITLE
[Due for payment 2025-02-18] [$250] Book Travel - Unable to continue setting up travel, error appears

### ISSUE THREAD

**Issue by IuliiaHerets** [Expensify/App#48305]
If you haven’t already, check out our [contributing guidelines](https://github.com/Expensify/ReactNativeChat/blob/main/contributingGuides/CONTRIBUTING.md) for onboarding and email contributors@expensify.com to request to join our Slack channel!
___

**Version Number:**  v9.0.26-4
**Reproducible in staging?:** Y
**Reproducible in production?:** Y 
Issue was found when executing this PR: https://github.com/Expensify/App/pull/47809
**Issue reported by:** Applause Internal Team

## Action Performed:
Go to https://staging.new.expensify.com/
Create a brand new account. 
Green + > Book Travel
Enter a company address -> Press `Save`
Press Book Travel -> Accept the terms of service -> Press `Continue`

## Expected Result:
Expensify Travel page opens

## Actual Result:
Error appears and Unable to continue setting 

## Workaround:
Unknown

## Platforms:

- [x] Android: Native
- [x] Android: mWeb Chrome
- [x] iOS: Native
- [x] iOS: mWeb Safari
- [x] MacOS: Chrome / Safari
- [x] MacOS: Desktop

## Screenshots/Videos

https://github.com/user-attachments/assets/80ecf02c-f920-499c-acbe-926cbaae39a0

</details>

[View all open jobs on GitHub](https://github.com/Expensify/App/issues?q=is%3Aopen+is%3Aissue+label%3A%22Help+Wanted%22)

<details><summary>Upwork Automation - Do Not Edit</summary>
    <ul>
        <li>Upwork Job URL: https://www.upwork.com/jobs/~01d6db9f1e6d7f6503</li>
        <li>Upwork Job ID: 1829616128762264680</li>
        <li>Last Price Increase: 2024-08-30</li>
<li>Automatic offers: </li>
<ul>
<li>shubham1206agra | Contributor | 105189846</li>
</ul></li>
    </ul>
</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @</details>

<details><summary>Issue Owner</summary>Current Issue Owner: @joekaufmanexpensify</details>

**Comment by IuliiaHerets** [comment]
We think that this bug might be related to #vip-travel

**Comment by IuliiaHerets** [comment]
@joekaufmanexpensify  FYI I haven't added the External label as I wasn't 100% sure about this issue. Please take a look and add the label if you agree it's a bug and can be handled by external contributors

**Comment by joekaufmanexpensify** [comment]
Discussing more [here](https://expensify.slack.com/archives/C05S5EV2JTX/p1725023725037389)

**Comment by joekaufmanexpensify** [comment]
@IuliiaHerets I got the `applause.expensifail.com` domain added to the beta. Going to close this for now as I don't think there is a bug here, but please reopen if you're still experiencing an issue after retesting!

**Comment by IuliiaHerets** [comment]
@joekaufmanexpensify Issue still reproducible. Build - v9.0.26-4, account - applausetester+ihchat4_3008_appa@applause.expensifail.com


https://github.com/user-attachments/assets/50b2cae0-9693-492d-9132-bd76e1764eb4

**Comment by joekaufmanexpensify** [comment]
@IuliiaHerets can you try again in ~30 minutes? It can sometimes take an hour+ for the beta to take effect.

**Comment by IuliiaHerets** [comment]
@joekaufmanexpensify QA team noticed that the issue can be reproduced with fresh accounts, after refreshing the page it works well. Build - v9.0.26-5, account - applausetester+ihchat4_3008_123@applause.expensifail.com.


https://github.com/user-attachments/assets/a306c4e4-4e21-4233-8067-3976e1f6cc41

**Comment by joekaufmanexpensify** [comment]
Ah, yeah. I can reproduce it when I do this with a brand new account. And then no longer can after refreshing. That said, we should still fix this as feels reasonable that some people will try to use travel soon after creating their accounts. 

https://github.com/user-attachments/assets/05b574b5-d77a-4b5a-9392-087d9cf5930c

**Comment by joekaufmanexpensify** [comment]
@Nodebrute are you thinking the issue here is that the beta isn't kicking in for a new account right away or something like that? As i experienced this issue, but my account is on the beta.

**Comment by joekaufmanexpensify** [comment]
@mananjadhav curious if you agree that this is likely a backend issue?

**Comment by joekaufmanexpensify** [comment]
Bump @mananjadhav

**Comment by mananjadhav** [comment]
Sorry I missed this one. I'll check this in a few hours.

**Comment by mananjadhav** [comment]
@joekaufmanexpensify I am not on beta. I've requested for the access. @joekaufmanexpensify can you confirm if the user is on the beta and you attempted to book travel? What is the API error you see for `AcceptSpotnanaTerms`.

**Comment by joekaufmanexpensify** [comment]
@mananjadhav yep, my account is on the beta. [Internal logs](https://www.expensify.com/_devportal/tools/logSearch/#sort=asc&size=10000&query=blob%3A%22AcceptSpotnanaTerms%22%20AND%20email%3A%22jkaufman%2Btraveltest2%40expensifail.com%22%20AND%20timestamp%3A%5B2024-09-05T00%3A00%20TO%202024-09-06T23%3A59%5D) for it are here. 

Error I see is `412 Can't provision Travel for the default workspace`.

**Comment by joekaufmanexpensify** [comment]
Thinking this is an internal bug, but LMK if you disagree

**Comment by mananjadhav** [comment]
Yeah that looks like an internal bug.

**Comment by joekaufmanexpensify** [comment]
Searching for a volunteer [here](https://expensify.slack.com/archives/C05S5EV2JTX/p1725549036289039).

**Comment by joekaufmanexpensify** [comment]
Pending volunteer

**Comment by joekaufmanexpensify** [comment]
Still pending internal volunteer

**Comment by joekaufmanexpensify** [comment]
Same. This is `High` in the project, so I expect it will get a volunteer fairly soon

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Still pending internal volunteer. Not too many high issues, so I expect it should get someone soon

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Pending internal volunteer

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
[Raised in slack again ](https://expensify.slack.com/archives/C05S5EV2JTX/p1727121516079899)

**Comment by joekaufmanexpensify** [comment]
Changing to monthly as NewDot travel is not the priority right now.

**Comment by joekaufmanexpensify** [comment]
Held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by stitesExpensify** [comment]
FYI this is off hold!

**Comment by mananjadhav** [comment]
Can this be worked externally?

**Comment by stitesExpensify** [comment]
I think this is an internal issue

**Comment by mananjadhav** [comment]
Thanks for confirming.

**Comment by joekaufmanexpensify** [comment]
@stitesExpensify do you plan to take this one or should I flag again in slack to find another volunteer?

**Comment by stitesExpensify** [comment]
I just posted today in the travel channel as part of the weekly update, I wouldn't be surprised if @cristipaval plans on picking this one up too :)

**Comment by stitesExpensify** [comment]
Thank you for offering though!

**Comment by joekaufmanexpensify** [comment]
Cool cool, sounds good!

**Comment by joekaufmanexpensify** [comment]
Still waiting for volunteer

**Comment by rlinoz** [comment]
I think this error is related to not setting the `testAccount` NVP before accepting the terms, at least I could not reproduce if I set the NVP to the new account.

Steps I took:
1. Create a new accoutn in NewDot
2. Go to OldDot and set the NVp
3. Go back to NewDot
4. Create a workspace
5. Go through the book travel flow

**Comment by joekaufmanexpensify** [comment]
@rlinoz I can still reproduce it with running the snippet if you try and book travel prior to creating a workspace in NewDot. Are you seeing the same?

**Comment by joekaufmanexpensify** [comment]
All good!

**Comment by rlinoz** [comment]
So it looks like the problem is because we try to provision travel in the personal policy, which is the user's default if they don't have any policy created yet.

Should we create something like we did for the Categorize expense flow to upgrade the user's accoutn?
<img width="2560" alt="Screenshot 2024-12-03 at 15 23 11" src="https://github.com/user-attachments/assets/28a7abe2-0017-41c4-9dac-ba9a9481cebf">

For reference this is what we show in OldDot

<img width="2071" alt="Screenshot 2024-12-03 at 15 21 41" src="https://github.com/user-attachments/assets/24d00642-187d-4ac9-90e3-6c44aedacaed">

**Comment by joekaufmanexpensify** [comment]
Got it. Yeah, something like that would make sense to me.

**Comment by rlinoz** [comment]
Cool, I agree, we will be able to make it external then and rely on this PR https://github.com/Expensify/App/pull/52737 that introduced the changes for categorization.

I can't really edit Figma, but it would look something like this with the proper texts:

![image](https://github.com/user-attachments/assets/90d049c6-3a9f-45bd-893e-5d2f38d0a73a)

cc: @Expensify/design

**Comment by dannymcclain** [comment]
That flow makes sense to me too 👍

**Comment by shawnborton** [comment]
That seems super straightforward - to basically reuse the upgrade pattern for Travel as well. Is this what you are thinking?
![CleanShot 2024-12-03 at 14 43 44@2x](https://github.com/user-attachments/assets/9d71e918-43d2-474c-b327-9cf9d938b59c)

**Comment by joekaufmanexpensify** [comment]
Yeah that looks solid. Only thing I'd change is that you just need to create any type of group workspace to use it (collect or control), so maybe the copy and button references "new workspace" instead of upgrade?

**Comment by shawnborton** [comment]
We currently just use "Upgrade" for the categories flow, so I'd be fine just reusing that:
![CleanShot 2024-12-03 at 15 08 13@2x](https://github.com/user-attachments/assets/19a4b895-4c12-48aa-84fa-4bd7e588cc4b)

**Comment by dannymcclain** [comment]
Yeah I think we should stick with the upgrade language as well and basically just copy the categories flow.

**Comment by joekaufmanexpensify** [comment]
Ah, I didn't realize that copy above was for a situation with no workspace. In that case, sounds good to me!

**Comment by cristipaval** [comment]
Taking this over, after chatting with @rlinoz.

The next steps are:
- Make the issue external and hire a contributor to work on that page.
- In parallel, fix the backend to throw a more specific error

**Comment by cristipaval** [comment]
> So we'll use the same issue for FE as well as BE?

yes, is there any downside?

**Comment by mananjadhav** [comment]
> > So we'll use the same issue for FE as well as BE?
> 
> yes, is there any downside?

Nope just clarifying :)

**Comment by joekaufmanexpensify** [comment]
@shubham1206agra is there an ETA for the FE PR here?

**Comment by cristipaval** [comment]
Could you please post an update, @shubham1206agra?

**Comment by cristipaval** [comment]
> That seems super straightforward - to basically reuse the upgrade pattern for Travel as well. Is this what you are thinking? ![CleanShot 2024-12-03 at 14 43 44@2x](https://private-user-images.githubusercontent.com/2319350/392113754-9d71e918-43d2-474c-b327-9cf9d938b59c.png?jwt=eyJhbGciOiJIUzI1NiIsInR5cCI6IkpXVCJ9.eyJpc3MiOiJnaXRodWIuY29tIiwiYXVkIjoicmF3LmdpdGh1YnVzZXJjb250ZW50LmNvbSIsImtleSI6ImtleTUiLCJleHAiOjE3MzM3NTE0NDcsIm5iZiI6MTczMzc1MTE0NywicGF0aCI6Ii8yMzE5MzUwLzM5MjExMzc1NC05ZDcxZTkxOC00M2QyLTQ3NGMtYjMyNy05Y2Y5ZDkzOGI1OWMucG5nP1gtQW16LUFsZ29yaXRobT1BV1M0LUhNQUMtU0hBMjU2JlgtQW16LUNyZWRlbnRpYWw9QUtJQVZDT0RZTFNBNTNQUUs0WkElMkYyMDI0MTIwOSUyRnVzLWVhc3QtMSUyRnMzJTJGYXdzNF9yZXF1ZXN0JlgtQW16LURhdGU9MjAyNDEyMDlUMTMzMjI3WiZYLUFtei1FeHBpcmVzPTMwMCZYLUFtei1TaWduYXR1cmU9NzU5OWNlMjdhMDE2NzQyNmE3Y2NhZWU0OGY0MGM0MjgyNTMwMGRmNzIzZmYzNmJjNjUwMDE1ZjUyNGJlYjNmYyZYLUFtei1TaWduZWRIZWFkZXJzPWhvc3QifQ.UyIjMrji5R65En9inrIGug5p-KOsmjWAunxaa81moLc)

We also allow the users to enable Travel on Collect workspaces. I think we should update the second paragraph of the description and make it similar to the page related to Categories. cc @shubham1206agra

**Comment by joekaufmanexpensify** [comment]
@shubham1206agra Is everything we need to move the PR forward resolved?

**Comment by shawnborton** [comment]
Can you add updated screens to the PR and tag me please? I think we have given copy suggestions in the PR and we're waiting for new screenshots. Thanks!

**Comment by joekaufmanexpensify** [comment]
Great. TY!

**Comment by cristipaval** [comment]
We're waiting on @shubham1206agra App PR.

**Comment by cristipaval** [comment]
What's the status here, @shubham1206agra?

**Comment by joekaufmanexpensify** [comment]
Got it. Putting on hold for that one. Will unhold once that PR is out.

**Comment by joekaufmanexpensify** [comment]
As an fyi, I am OOO until the new year. Please ask in slack if anything urgent BZ related comes up. Otherwise, I will continue assisting when I return!

**Comment by joekaufmanexpensify** [comment]
Still held

**Comment by joekaufmanexpensify** [comment]
Same

**Comment by joekaufmanexpensify** [comment]
Still blocked on the above PR. Hope to be unblocked soon.

**Comment by cristipaval** [comment]
The PR just got merged. Should get deployed today.

**Comment by joekaufmanexpensify** [comment]
Blocking PR is out to prod. @shubham1206agra @cristipaval are we good to un-hold this and finish up this issue's PR?

**Comment by cristipaval** [comment]
Removed the hold. @shubham1206agra, what's the status here?

**Comment by joekaufmanexpensify** [comment]
TY!

**Comment by joekaufmanexpensify** [comment]
PR out on staging

**Comment by mananjadhav** [comment]
Not overdue. This will be paid out tomorrow.

**Comment by joekaufmanexpensify** [comment]
@mananjadhav can you handle checklist as part of prepping for payment?

**Comment by mananjadhav** [comment]
@joekaufmanexpensify I think this is more of a feature request. This started as a backend bug and ended up becoming a feature request. Adding the checklist nevertheless and I think we should add a regression test for the same.

# BugZero Checklist:

- [x] **[Contributor]** Classify the bug:

<details>
<summary>Bug classification</summary>


Source of bug:
  - [x] 1a. Result of the original design (eg. a case wasn't considered)
  - [ ] 1b. Mistake during implementation
  - [ ] 1c. Backend bug
  - [ ] 1z. Other:

Where bug was reported:
  - [x] 2a. Reported on production (eg. bug slipped through the normal regression and PR testing process on staging)
  - [ ] 2b. Reported on staging (eg. found during regression or PR testing)
  - [ ] 2d. Reported on a PR
  - [ ] 2z. Other:

Who reported the bug:
  - [ ] 3a. Expensify user
  - [ ] 3b. Expensify employee
  - [ ] 3c. Contributor
  - [x] 3d. QA
  - [ ] 3z. Other:

</details>

- [x] **[Contributor]** The offending PR has been commented on, pointing out the bug it caused and why, so the author and reviewers can learn from the mistake.

    Link to comment: NA

- [x] **[Contributor]** If the regression was CRITICAL (e.g. interrupts a core flow) A discussion in [#expensify-open-source](https://app.slack.com/client/E047TPA624F/C01GTK53T8Q) has been started about whether any other steps should be taken (e.g. updating the PR review checklist) in order to catch this type of bug sooner.

    Link to discussion: NA

- [x] **[Contributor]** If it was decided to create a regression test for the bug, please propose the [regression test](https://github.com/Expensify/App/blob/main/contributingGuides/REGRESSION_TEST_BEST_PRACTICES.md) steps using the template below to ensure the same bug will not reach production again.

- [ ] **[BugZero Assignee]** Create a GH issue for creating/updating the regression test once above steps have been agreed upon.

    Link to issue:

## Regression Test Proposal
### Precondition:

- Have a new account with `spotnanaTravel` beta enabled and no workspace created.

### Test:
 1. Click on FAB  Book Travel.
 2. Click on Book travel.
 3. Verify that upgrade flow has started.
 4. Click on Upgrade.
 5. Verify that workspace confirmation page appears.
 6. Fill all the information
 7. Click on Confirm.
 8. Verify that upgrade confirmation screen appears.

Do we agree 👍 or 👎

**Comment by joekaufmanexpensify** [comment]
Regression test issue [here](https://github.com/Expensify/Expensify/issues/471710). Checklist complete.

**Comment by joekaufmanexpensify** [comment]
All set to issue payment. We need to pay: 

- @shubham1206agra - $250 for PR via Upwork. 
- @mananjadhav - $250 for C+ via NewDot.

**Comment by joekaufmanexpensify** [comment]
@shubham1206agra your original offer expired, so I created a [new upwork job](https://www.upwork.com/jobs/~021891854674303189456) and sent you an offer. Please accept it so we can pay you.

**Comment by joekaufmanexpensify** [comment]
@mananjadhav please request payment at your earliest convenience.

**Comment by mananjadhav** [comment]
I just did. Thanks.

**Comment by joekaufmanexpensify** [comment]
@shubham1206agra $250 sent and contract ended!

**Comment by joekaufmanexpensify** [comment]
Upwork job closed.

**Comment by joekaufmanexpensify** [comment]
All set, thanks everyone!
